src/test/scala/firrtlTests/transforms/GroupComponentsSpec.scala

```diff
@@ -0,0 +1,290 @@
+         |    out <= UInt(1)
+         |  module Const2A :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+         |  module Const1B :
+         |    output out: UInt<8>
+         |    out <= UInt(1)
+         |  module Const2B :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+      """.stripMargin
+    val groups = Seq(
+      GroupAnnotation(Seq(topComp("c1a"), topComp("c2a"), topComp("asum")), "A", "cA", Some("_OUT"), Some("_IN")),
+      GroupAnnotation(Seq(topComp("c1b"), topComp("c2b"), topComp("bsum")), "B", "cB", Some("_OUT"), Some("_IN"))
+    )
+    val check =
+      s"""circuit Top :
+         |  module $top :
+         |    output out: UInt<16>
+         |    inst cA of A
+         |    inst cB of B
+         |    out <= add(cA.asum_OUT, cB.bsum_OUT)
+         |  module A :
+         |    output asum_OUT: UInt<9>
+         |    inst c1a of Const1A
+         |    inst c2a of Const2A
+         |    node asum = add(c1a.out, c2a.out)
+         |    asum_OUT <= asum
+         |  module B :
+         |    output bsum_OUT: UInt<9>
+         |    inst c1b of Const1B
+         |    inst c2b of Const2B
+         |    node bsum = add(c1b.out, c2b.out)
+         |    bsum_OUT <= bsum
+         |  module Const1A :
+         |    output out: UInt<8>
+         |    out <= UInt(1)
+         |  module Const2A :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+         |  module Const1B :
+         |    output out: UInt<8>
+         |    out <= UInt(1)
+         |  module Const2B :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+      """.stripMargin
+    execute(input, check, groups)
+  }
+
+  "The two sets of instances" should "be grouped with one not grouped" in {
+    val input =
+      s"""circuit $top :
+         |  module $top :
+         |    output out: UInt<16>
+         |    inst c1a of Const1A
+         |    inst c2a of Const2A
+         |    inst c1b of Const1B
+         |    inst c2b of Const2B
+         |    node asum = add(c1a.out, c2a.out)
+         |    node bsum = add(c1b.out, c2b.out)
+         |    inst pass of PassThrough
+         |    pass.in <= add(asum, bsum)
+         |    out <= pass.out
+         |  module Const1A :
+         |    output out: UInt<8>
+         |    out <= UInt(1)
+         |  module Const2A :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+         |  module Const1B :
+         |    output out: UInt<8>
+         |    out <= UInt(1)
+         |  module Const2B :
+         |    output out: UInt<8>
+         |    out <= UInt(2)
+         |  module PassThrough :
+         |    input in: UInt
+         |    output out: UInt
+         |    out <= in
+      """.stripMargin
+    val groups = Seq(
+      GroupAnnotation(Seq(topComp("c1a"), topComp("c2a"), topComp("asum")), "A", "cA", Some("_OUT"), Some("_IN")),
+      GroupAnnotation(Seq(topComp("c1b"), topComp("c2b"), topComp("bsum")), "B", "cB", Some("_OUT"), Some("_IN"))
```

Comment:
If the suffixes are allowed to be `None`, I'd like to see an example with that.

Reply:
An example is beneath, line 253.